### PR TITLE
[DNM] Bug 1397725 - Add context to newrelic transactions.

### PR DESCRIPTION
**Pull request by adngdb**
@mathjazz I am not really confident about how to test this. I see we have newrelic set up for our staging environment, I guess that would be the best way to test it? What do you think? 

The goal of this is to get more insights about those traces we have for the entities page, and be able to better understand what path is slow in our code. Also, it's a first step making it easier to add more context later on.

### CODE REVIEW

File: pontoon/base/utils.py [hunk 1]
```diff
@@ -653,3 +659,14 @@ def build_translation_memory_file(creation_date, locale_code, entries):
...
+
+
+def add_newrelic_context(**kwargs):
+    try:
+        for key, value in kwargs.items():
+            newrelic.agent.add_custom_parameter(key, value)
+    except AttributeError:
+        # If newrelic-admin isn't used, for example in local development,
+        # then newrelic.agent will not be set.
+        log.warning('Unable to add context to newrelic')
```
Comment (mathjazz) [inline pontoon/base/utils.py]: This might get a little bit too verbose over time once we add more context?

File: pontoon/base/views.py [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import logging
 import os
+from datetime import datetime
 
 from bulk_update.helper import bulk_update
-from datetime import datetime
```
Comment (mathjazz) [inline pontoon/base/views.py]: I'm curious, why did you move this?

File: pontoon/base/utils.py (hunk 1 above)
Comment (adngdb) [reply pontoon/base/utils.py]: That should only happen in environments where we do not have newrelic enabled, and as far as I know, that means only dev. I guess we can do with excessive amounts of logs in dev?

File: pontoon/base/views.py (hunk 2 above)
Comment (adngdb) [reply pontoon/base/views.py]: Well ``datetime`` is part of the standard library, so it should be with the other imports from that standard library, at the top.

File: pontoon/base/utils.py (hunk 1 above)
Comment (mathjazz) [reply pontoon/base/utils.py]: OK, let's keep it.

File: pontoon/base/views.py (hunk 2 above)
Comment (mathjazz) [reply pontoon/base/views.py]: Oh, OK. For some reason I got misled by the `from` into thinking this is not a standard library, but `form` has nothing to do with it of course.

**Comment by adngdb** [comment]
This is not working yet. I'm trying to figure out how we set up NewRelic in heroku, to understand what might be wrong. I suspect we have some incompatibilities around the moving parts related to NewRelic.

**Comment by adngdb** [comment]
I'm still struggling with this, I've [asked for help on NewRelic's forum](https://discuss.newrelic.com/t/newrelic-has-no-attribute-agent-with-newest-versions/51867).

**Comment by adngdb** [comment]
Closing this for now. I can't make any progress here without help from NewRelic, and they don't seem to be willing to help, sadly. Also, this is not as important as I thought it was.
